### PR TITLE
Rework CPU MeanVarianceNormalization kernel to support arbitrary axes.

### LINKED ISSUE #1828: MeanVarianceNormalization v9 CPU kernel doesn't honor axes attribute
**Describe the bug**
The MeanVarianceNormalization v9 CPU kernel doesn't honor the `axes` attribute, which supports arbitrary axes. It just tries to map them into the *older* operator kernel's parameters (the `cross_channel` attribute, which supported only NCHW and NHW). However, the mapping doesn't do any error checking to see whether the `axes` are actually mappable (e.g. HW gets blindly mapped to NHW, or NC gets mapped to NCHW, which isn't correct). So the core issue is that v9 isn't fully implemented for reduction along arbitrary axes, but a bigger issue is that it silently returns the wrong result - it should return a failing error message for incompatible parameters.

https://github.com/microsoft/onnxruntime/blob/8712a523a4cda4500256aafae378558e64d1f672/onnxruntime/core/providers/cpu/tensor/mean_variance_normalization.h#L108

**Urgency**
When able. We haven't found any models locally that actually use MVN (scanning 1381 in the WinML model collection), excluding our ONNX conformance tests.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 Vibranium
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: 0.5
- Python version: NA
- Visual Studio version (if applicable): 2017
- GCC/Compiler version (if compiling from source): NA
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**To Reproduce**
Pass `axes` with [2,3] to MVN v9:

    {
      "op_type": "MeanVarianceNormalization",
      "version": 9,
      "axes": [2,3], // Only height, width (no normalization across batch dimension)
      "X": {"value": [ [[[1, 2], [3, 4]]], [[[1, 2], [3, 4]]] ]},
      "Y": {"value": [ [[[-1.34164, -0.447214], [0.447214, 1.34164]]], [[[-1.34164, -0.447214], [0.447214, 1.34164]]] ]},
      "T": "float32"
    }

**Expected behavior**
Mean reduction occurs along `axes` passed.

Until then, report an error:

    constexpr int64_t cross_channel_axes[] = { 0,1,2,3 };
    constexpr int64_t batch_spatial_axes[] = { 0,2,3 };

    if (std::equal(std::begin(axes), std::end(axes), std::begin(cross_channel_axes), std::end(cross_channel_axes)))
    {
        this->across_channels_ = true;

    }
    else if (std::equal(std::begin(axes), std::end(axes), std::begin(batch_spatial_axes), std::end(batch_spatial_axes)))
    {
        this->across_channels_ = false;
    }
    else
    {
        ORT_THROW("MeanVarianceNormalization axes parameter only supports NHW and NCHW reduction.");
    }

**Additional context**
This was found during ONNX conformance test testing for WinML.

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

Rework CPU MeanVarianceNormalization kernel to support arbitrary axes.

### Motivation and Context

Fix #1828.

### CODE REVIEW

File: onnxruntime/test/providers/cpu/tensor/tensor_op_test.cc [hunk 1]
```diff
@@ -489,35 +489,31 @@ void MeanVarianceNormalizationFunctionDefaultPerChannel() {
...
-  // axes: {0, 1, 2, 3} for across_channels
-  MeanVarianceNormalizationFunctionAcrossChannels({0, 1, 2, 3});
-
-  // Default (axes: {0, 2, 3}) for non across_channels
-  MeanVarianceNormalizationFunctionDefaultPerChannel();
+TEST(TensorOpTest, MeanVarianceNormalizationAllAxes) {
+  TestMeanVarianceNormalizationOverAllAxes({2, 2, 4});
+  TestMeanVarianceNormalizationOverAllAxes({2, 2, 2, 3});
+  TestMeanVarianceNormalizationOverAllAxes({2, 2, 2, 2, 2});
 }
```
Comment (edgchen1) [inline onnxruntime/test/providers/cpu/tensor/tensor_op_test.cc:517]: add test for some other axes subset

File: onnxruntime/core/providers/cpu/tensor/mean_variance_normalization.cc [hunk 2]
```diff
@@ -3,17 +3,206 @@
...
+    : OpKernel{info},
+      normalize_variance_{
+          // legacy attribute
+          info.GetAttrOrDefault("normalize_variance", int64_t{1}) == int64_t{1}},
+      axes_{GetAxesFromAttribute(info)} {
+}
+
+Status MeanVarianceNormalization::Compute(OpKernelContext* context) const {
+  const auto& input = context->RequiredInput<Tensor>(0);
+  const auto input_shape = input.Shape();
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/tensor/mean_variance_normalization.cc:128]: nit: const auto&

File: onnxruntime/test/providers/cpu/tensor/gen_mvn_test_data.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import argparse
```
Comment (pranavsharma) [inline onnxruntime/test/providers/cpu/tensor/gen_mvn_test_data.py:1]: Missing license/copyright header

**Review by pranavsharma — APPROVED** [review]

**Comment by fdwr** [comment]
Thanks @edgchen1 - now I can finally re-enable that test case for both DML *and* CPU. :D

**Pull request merged**